website/project/views/node.py

```diff
@@ -1008,6 +1008,66 @@ def get_children(auth, node, **kwargs):
     return _render_nodes(nodes, auth)
 
 
+def node_child_tree(user, node_ids):
```

Comment:
Node logs have some handy functions for gathering the tree structure. If nothing else, you might be able to use one of those to start with.